src/Commands/Workflow/WatchCommand.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Workflow;
+
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Pantheon\Terminus\Site\SiteAwareInterface;
+use Pantheon\Terminus\Site\SiteAwareTrait;
+
+define('WORKFLOWS_WATCH_INTERVAL', 5);
+
+class WatchCommand extends TerminusCommand implements SiteAwareInterface
+{
+    use SiteAwareTrait;
+
+    /**
+     * Streams new and finished workflows to the console
+     *
+     * @command workflow:watch
+     *
+     * @param string $site_id Site name to watch workflows on.
+     *
+     * @usage terminus workflow:watch <site_name>
+     *   Watch workflows on site <site_name>
+     */
+    public function watch($site_id)
+    {
+        $site = $this->getSite($site_id);
+        $date_format = $this->getConfig()->get('date_format');
+
+        // Keep track of workflows that have been printed.
+        // This is necessary because the local clock may drift from
+        // the server's clock, causing events to be printed twice.
+        $started  = [];
+        $finished = [];
+
+        $this->log()->notice('Watching workflows...');
+        $site->workflows->fetchWithOperations();
+        while (true) {
```

Comment:
I would prefer these be optimized and I've been fixing things around my commands as the opportunities present themselves.